src/Aquifer.Data/EventHandlers/BadTranslationPairHandler.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Aquifer.Data.Entities;
+using Aquifer.Data.Exceptions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace Aquifer.Data.EventHandlers;
+
+public static class BadTranslationPairHandler
+{
+    private static readonly IReadOnlyList<string> s_KeyBlacklist = ["span", "div"];
+
+    public static void Handle(IEnumerable<EntityEntry> entityEntries)
+    {
+        entityEntries.Where(entry => entry is { Entity: TranslationPairEntity, State: EntityState.Modified or EntityState.Added })
+            .Select(x => x.Entity as TranslationPairEntity)
+            .ToList()
```

Comment:
Is `ToList()` needed?

Reply:
Yup

Reply:
Oh, interesting.  It only works on a `List<T>` and not an `IEnumerable<T>`.  I have historically avoided `ForEach()` like the plague and prefer to just do:
```
foreach (var item in Chain().Of().Linq().Commands())
{
    DoTheThing();
}
```
This makes it clear that a mutation is happening (otherwise we'd use a `Select()`).  It also allows for the removal of the `ToList()` call as a micro-optimization.